src/platform/unix/mod.rs

```diff
@@ -415,16 +423,16 @@ impl OsIpcChannel {
 
 pub struct OsIpcReceiverSet {
     incrementor: Incrementor,
-    pollfds: Vec<pollfd>,
-    fdids: HashMap<c_int, u64, BuildHasherDefault<FnvHasher>>
+    poll: Poll,
+    pollfds: HashMap<Token, PollEntry, BuildHasherDefault<FnvHasher>>,
+    events: Events
 }
 
 impl Drop for OsIpcReceiverSet {
     fn drop(&mut self) {
-        unsafe {
-            for pollfd in self.pollfds.iter() {
-                let result = libc::close(pollfd.fd);
-                assert!(thread::panicking() || result == 0);
+        for &PollEntry { id: _, fd } in self.pollfds.values() {
+            unsafe {
+                assert!(thread::panicking() || libc::close(fd) == 0);
```

Comment:
I'm sorry to say that this is still not right :-( Again, this code won't even try to close the FDs during a panic, which is wrong. (And also, putting actual functionality in an `assert!()` rather than just checks is not a good idea.)

Really, there are good reasons why this code is arranged this way...